internal/internal.go

```diff
@@ -207,9 +207,10 @@ var (
 	// default resolver scheme.
 	UserSetDefaultScheme = false
 
-	// ShuffleAddressListForTesting pseudo-randomizes the order of addresses.  n
+	// ShuffleAddressListForTesting pseudo-randomizes the order of addresses. n
 	// is the number of elements.  swap swaps the elements with indexes i and j.
-	ShuffleAddressListForTesting any // func(n int, swap func(i, j int))
+	// It returns the shuffle func before it was replaced.
+	ShuffleAddressListForTesting any // func(sf func(n int, swap func(i, j int))) func(n int, swap func(i, j int))
```

Comment:
I think we should move this and the other variable that I suggested in the other comment to live in `balancer/pickfirst/internal` instead of in the top-level `internal` package. This would also mean that we would need to move the test from `test/pickfirst_test.go` to `balancer/pickfirst/pickfirst_ext_test.go` (and the package name would `pickfirst_test`).

Not so long ago, the pickfirst LB policy was in the top-level `grpc` package and therefore this variable living in `internal` made sense and the test living in the top-level `test` directory made sense too. Now that we have a pickfirst LB policy in a separate package, we should move these over there too.

I'm fine if you want to do these changes in a follow-up PR or right here. 

Thanks.

Reply:
The pickfirst test may as well live inside the pickfirst directory. Is there any specific reason for it to live in `grpc/test/...` other than the historical reason that pickfirst was originally part of the `grpc` package?

Reply:
Also helps with overall runtimes if tests are spread across packages instead of crowding inside the top-level `test` package.

Reply:
> The pickfirst test may as well live inside the pickfirst directory. Is there any specific reason for it to live in grpc/test/... other than the historical reason that pickfirst was originally part of the grpc package?

If it can be moved easily, then I'm fine with that to make this even simpler.

We do have a billion tests in `grpc/test` that have deeply intertwined dependencies, and I'm not sure whether this is one of them.

Reply:
This is a more recently added test file, compared to much of the ancient stuff in `grpc/test`. I skimmed through it and it does not look like it is using any of the utilities that are part of the `grpc/test` package. So, it should be fairly easy to move it. But I'll leave it to @arjan-bal to decide.

Reply:
Moved the files into `/balancer/pickfirst`.